apis/logging/v1/output_types.go

```diff
@@ -277,3 +279,24 @@ type Splunk struct {
 	// +optional
 	Fields []string `json:"fields,omitempty"`
 }
+
+// Http provided configuration for sending logs to a generic http endpoint
+type Http struct {
+	// Encoding to be used for sending logs.
+	//
+	// +kubebuilder:validation:Enum:=json
+	// +optional
+	Encoding string `json:"encoding,omitempty"`
```

Comment:
Don't need this yet - we can add it later if/when we have more than one encoding.

Reply:
> Add a "method" and "maxBatch" as per https://docs.google.com/document/d/1VdeOohi70ZQdvJggxYZbxBp7vPTZ4XMVDOBh8ZYvtaw/edit#

Do we want `maxBatch` in terms of bytes, or number of events? vector supports 3 kinds
https://v0-23.vector.dev/docs/reference/configuration/sinks/http/#batch

Reply:
Bytes. The limits on loki or the like are in bytes and the size of events is not predictable.

Reply:
@vimalk78 does fluent support "max" bytes in this context.  The choice will need to apply to both

Reply:
`fluentd` doesn't seem to support any config parameters for batching.  but recommends putting multiple records in same message with json
https://docs.fluentd.org/input/http#handle-large-data-with-batch-mode